src/components/Story-9-AddBook.jsx

```diff
@@ -0,0 +1,150 @@
+import { useEffect, useState } from "react"
+
+export default function AddBook() {
+
+    const [genres, setGenres] = useState([])
+
+    const [title, setTitle] = useState("")
+    const [author, setAuthor] = useState("")
+    const [genre, setGenre] = useState(null)
+    const [year, setYear] = useState(NaN)
+    const [pageCount, setPageCount] = useState("")
+    const [imageURL, setImageURL] = useState("") // declare placeholder image if user doesnt input an image url
+    const [blurb, setBlurb] = useState("")
+    
+    const [postBody, setPostBody] = useState({})
+
+    const [errors, setErrors] = useState({})
+
+    const [success, setSuccess] = useState(false)
+
+    function fetchGenres(){
+        fetch('https://book-swap-api.dev.io-academy.uk/api/genres', {
+            mode: 'cors',
+            headers: {
+                "Content-Type": "application/json",
+                "Accept": "application/json"
+            }
+        })
+            .then(res => res.json())
+            .then(data => {
+                setGenres(data.data)
+                setGenre(0)
+        })
+    }
+
+    function submitBook(e) {
+        e.preventDefault()
+        try{
+            fetch('https://book-swap-api.dev.io-academy.uk/api/books', {
+                mode: 'cors',
+                method : 'POST',
+                headers: {
+                    "Content-Type": "application/json",
+                    "Accept": "application/json"
+                },
+                body : JSON.stringify(postBody)
+            })
+                .then(res => {
+                    if (res.status == 201) {
+                        setSuccess(true)
+                    }
+                    return res.json()
+            })
+                .then(data => {
+                    if (data.errors){
+                        setErrors(data.errors)
+                    }
+            })
+        } catch(err) {
+            console.error(err)
+        }
+    }
+
+    function createPostBody(){
+        let body = {}
+
+        if (title) body.title = title
+        if (author) body.author = author
+        if (genre !== null) body.genre_id = genre
+        if (!isNaN(year)) body.year = year
+        if (blurb) body.blurb = blurb
+        if (imageURL) body.image = imageURL
+
+        setPostBody(body)
```

Comment:
No need to use state for this, or to have it connected with a useEffect.

I would instead set this function up so it just returns body, and then you can call it directly in the submitBook function